✨ feat: Add switch for including historical reasoning in context (Interleaved thinking)

sxjeru opened the pull request:
#### 💻 Change Type

<!-- For change type, change [ ] to [x]. -->

- [ ] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] ✅ test
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔗 Related Issue

- #9769
- #9438

#### 🔀 Description of Change

- https://platform.minimaxi.com/docs/guides/text-m2-function-call
- https://platform.moonshot.cn/docs/guide/use-kimi-k2-thinking-model

上述需独立适配。

受 https://linux.do/t/topic/1109582 启发，支持将历史消息中的推理内容添加到上下文中一并发送给大模型，维持完整思维链路。

开关添加在 会话设置 -> 模型设置 末尾。目前仅会影响类 openai 的 ChatCompletion 端口。（排除 Anthropic、Google 和 VertexAI）

<img width="963" height="329" alt="image" src="https://github.com/user-attachments/assets/cccec728-4332-4c7f-a690-bd587215f7e2" />


#### 🧪 How to Test

<!-- Please describe how you tested your changes -->

<!-- For AI features, please include test prompts or scenarios -->

- [ ] Tested locally
- [x] Added/updated tests
- [ ] No tests needed

#### 📸 Screenshots / Videos

<!-- If this PR includes UI changes, please provide screenshots or videos -->

| Before | After |
| ------ | ----- |
| ...    | ...   |

#### 📝 Additional Information

<!-- Add any other context about the Pull Request here. -->

<!-- Breaking changes? Migration guide? Performance impact? -->

## Summary by Sourcery

Introduce a toggle to include historical reasoning content in the message context for OpenAI-compatible chat models and propagate it through the context construction pipeline.

New Features:
- Add includeHistoricalThinking option in Agent settings UI under model settings
- Add includeHistoricalThinking flag to agent chat configuration, runtime payloads, and validation schema
- Embed historical reasoning content in messages using <think> tags when the switch is enabled

Enhancements:
- Propagate includeHistoricalThinking flag through contextEngineering service and ChatService pipeline

Documentation:
- Add localization entries for the includeHistoricalThinking setting label and description

## Summary by Sourcery

Add includeHistoricalThinking toggle and interleaved thinking support across the chat pipeline

New Features:
- Introduce includeHistoricalThinking switch in Agent Settings UI under model settings
- Add includeHistoricalThinking flag to agent chat config schema, ChatService, and contextEngineering pipeline
- Embed historical reasoning in messages as <think> tags for OpenAI-compatible providers when enabled
- Preserve raw reasoning fields for MiniMax and Moonshot providers when enabled
- Emit reasoning_details in MiniMax and reasoning_content in Moonshot runtimes for interleaved thinking
- Extend OpenAIStream to parse MiniMax reasoning_details streaming events
- Add AI model cards for Moonshot K2 Thinking (Turbo) and MiniMax M2 Stable

Enhancements:
- Refactor MessageCleanupProcessor to use includeHistoricalThinking and provider config
- Propagate includeHistoricalThinking through contextEngineering and ChatService pipelines

Documentation:
- Add localization entries for includeHistoricalThinking setting title and description

Tests:
- Add unit tests for MessageContentProcessor embedding logic across providers and configurations
- Add unit tests for MessageCleanupProcessor preserving or dropping reasoning fields
- Add unit tests for OpenAIStream handling MiniMax M2 reasoning_details format

sxjeru committed:
✨ feat: 添加历史思考内容选项，增强消息上下文理解

sxjeru committed:
add tests

arvinxx commented:
I suggest writing an RFC for this. It's a fairly straightforward feature that significantly impacts context handling. Also, I don't think Interleaved Thinking should be supported in this way - it should be implemented in model-runtime instead.

---
> This comment was translated by Claude.

<details>
<summary>Original Content</summary>
这个建议写个 RFC,是一个比较直接的显著上下文的 feature 。以及我觉得要支持 Interleaved Thinking 也不应该是这样支持,应该得在 model-runtime 里做
</details>

sxjeru committed:
Merge branch 'next' into 10.31

sxjeru committed:
✨ feat: 适配 MiniMax-M2
enhance MessageCleanup and MessageContent processors to support reasoning preservation for MiniMax

sxjeru commented:
<img width="1748" height="205" alt="image" src="https://github.com/user-attachments/assets/ab29824b-0055-4901-977b-b63d0e5ef42a" />

<img width="1762" height="208" alt="image" src="https://github.com/user-attachments/assets/63f905da-b34f-4885-b74d-6ccc67aa37c8" />

sxjeru committed:
✨ feat: 添加 Kimi K2 思考模型及其高速版本，增强聊天模型功能

sxjeru committed:
Update packages/model-runtime/src/core/streams/openai/openai.ts

Co-authored-by: sourcery-ai[bot] <58596630+sourcery-ai[bot]@users.noreply.github.com>

arvinxx commented:
@sxjeru For this PR, first make it only include the Interleaved Thinking mode for MiniMax, without modifying other parts. Later, DeepSeek's approach to passing back thinking will likely be different from this one. This way, we can get MiniMax's specialized thinking feature working first, and then this PR can be merged. I don't recommend adding other thinking pass-back functionality here, and I don't think this is the best practice either.

---
> This comment was translated by Claude.

<details>
<summary>Original Content</summary>
@sxjeru 这个 PR 先做成只包含 minimax 的 Interleaved Thinking 的模式,不要改动其他的部分。后面 DeepSeek 的回传 thinking 的方案估计也和这个不一样。这样先把 minimax 的这个特化 thinking 弄好这个PR就可以合了。其他的回传 Thinking 这个我不建议在这里加。而且我也不觉得这是最佳实践
</details>
